scalatest/src/main/scala/org/scalatest/prop/Randomizer.scala

```diff
@@ -280,47 +328,113 @@ class Randomizer(private[scalatest] val seed: Long) { thisRandomizer =>
     * @return A random positive Float, and the next Randomizer to use.
     */
   def nextPosZFloat: (PosZFloat, Randomizer) = {
-    val (f, r) = nextFloat
-    val pos = f.abs // 0.0f or greater
-    (PosZFloat.ensuringValid(pos), r)
+    // scala> Float.NegativeInfinity.abs
+    // res30: Float = Infinity
+    //
+    // scala> -0.0f.abs
+    // res31: Float = 0.0
+    //
+    // scala> PosZFloat(-0.0f)
+    // res0: org.scalactic.anyvals.PosZFloat = PosZFloat(-0.0f)
+    val (candidate, r) = nextExtRealFloatValue
+    val posZ =
+      if (isNegativeZeroFloat(candidate)) candidate // leave it negative zero
```

Comment:
In principle, this seems wrong.  Granted, the distinction between negative and positive zero is at the angels-dancing-on-a-pin level, but it feels inappropriate for PosZFloat to include negative zero; I would intuitively expect this to become positive zero...

Reply:
I know, but I didn't think of this before releasing PosZFloat and PosZDouble to the wild, and they allow -0.0. So changing that now would need to go through a deprecation cycle with renaming, which would be a pain. But also -0.0 is actually a Float zero. So it could go either way. PosInt means the postitive part of the space, PosZInt means the positive part of the space plus zero. It doesn't mean "positive zero," because there is no such thing. But let's keep thinking about it. It was surprising to me, and there is at least one way I know of that the -0.0. can be transformed to a non-zero value but still be negative:

```
scala> 1 / -0.0
res7: Double = -Infinity
```